.github/workflows/end-to-end-test.yml

```diff
@@ -22,120 +22,285 @@ env:
+    # In that case this Action is triggered in pull_request_target context and checks out the HEAD of the PR branch.
+    # This is a semi-secure manually reviewed way to ensure we only run code we're fine accessing our secrets
+    - name: Check out PR # nosemgrep:yaml.github-actions.security.pull-request-target-code-checkout.pull-request-target-code-checkout
+      if: github.event.action == 'labeled' && github.event.label.name == env.E2E_TEST_LABEL
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: 🏁 Build release binaries
+      if: github.repository == env.BASE_REPO && (github.event.action != 'labeled' || github.event.label.name == env.E2E_TEST_LABEL)
+      run: make build-release
+
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      checks: write
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe #v4.1.0
+      with:
+        go-version: '~1.20'
+      id: go
+
+    - name: Check out base repo
+      if: github.event.action != 'labeled'
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+
+    # If a PR was reviewed and deemed safe to run in the context of our repo and it's secrets, we label it to trigger E2E tests.
+    # In that case this Action is triggered in pull_request_target context and checks out the HEAD of the PR branch.
+    # This is a semi-secure manually reviewed way to ensure we only run code we're fine accessing our secrets
+    - name: Check out PR # nosemgrep:yaml.github-actions.security.pull-request-target-code-checkout.pull-request-target-code-checkout
+      if: github.event.action == 'labeled' && github.event.label.name == env.E2E_TEST_LABEL
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: 🌎 Integration test
+      if: github.repository == env.BASE_REPO && (github.event.action != 'labeled' || github.event.label.name == env.E2E_TEST_LABEL)
+      run: make integration-test testopts="--junitfile test-result-integration.xml"
+      env:
+        URL_ENVIRONMENT_1: ${{ secrets.URL_ENVIRONMENT_1 }}
+        URL_ENVIRONMENT_2: ${{ secrets.URL_ENVIRONMENT_2 }}
+        TOKEN_ENVIRONMENT_1: ${{ secrets.TOKEN_ENVIRONMENT_1 }}
+        TOKEN_ENVIRONMENT_2: ${{ secrets.TOKEN_ENVIRONMENT_2 }}
+        PLATFORM_URL_ENVIRONMENT_1: ${{ secrets.PLATFORM_URL_ENVIRONMENT_1 }}
+        PLATFORM_URL_ENVIRONMENT_2: ${{ secrets.PLATFORM_URL_ENVIRONMENT_2 }}
+        OAUTH_CLIENT_ID: ${{ secrets.OAUTH_CLIENT_ID }}
+        OAUTH_CLIENT_SECRET: ${{ secrets.OAUTH_CLIENT_SECRET }}
+        OAUTH_TOKEN_ENDPOINT: ${{ secrets.OAUTH_TOKEN_ENDPOINT }}
+
+    - name: 🧹 Cleanup
```

Comment:
! cleanup probably shouldn't run parallel to other jobs! It will purge the environments - which we certainly don't want to happen while other tests run. 

This should be removed from all but the new "nightly job" or be it's own thing - and we should limit that job to run after the other's have finished

Reply:
Yes, this version was the first 'let's see if this even works' version, the new one cleans it up correctly.